lib/internal/bootstrap_node.js

```diff
@@ -621,9 +634,31 @@
     nativeModule.cache();
     nativeModule.compile();
 
+    if (config.experimentalModules) {
+      const ns = NativeModule._nsCache[id] = Object.create(null);
+      if (typeof nativeModule.exports === 'function')
+        ns.default = nativeModule.exports;
+
+      const descriptors =
+        Object.getOwnPropertyDescriptors(nativeModule.exports);
+      for (const [name, d] of Object.entries(descriptors)) {
+        if (!d.enumerable || d.set !== undefined)
+          continue;
+        if (d.get !== undefined && !lazyDep().has(d.get))
+          ns[name] = (...args) => nativeModule.exports[name](...args);
+        else
+          ns[name] = nativeModule.exports[name];
+      }
+    }
+
```

Comment:
@devsnek Can you explain this bit:
```js
if (d.get !== undefined && !lazyDep().has(d.get))
  ns[name] = (...args) => nativeModule.exports[name](...args);
else
  ns[name] = nativeModule.exports[name];
```

Reply:
some methods are attached as getters which will fire when the method is accessed instead of called, this wraps it so that the getter is fired when it is called.

Reply:
Ah thanks!

Would `this` binding ever come into play?
The `ns[name] = nativeModule.exports[name]` no longer has the `exports` as its `this` while the wrapped form does.

Reply:
no builtin modules that i know of use that behavior

Reply:
Should the check be `lazyDep().has(d.get)` without the leading `!`?